### ISSUE TITLE
Automatically find ymin/ymax in plots

### ISSUE THREAD

**Issue by jcohenadad** [courtois-neuromod/anat-processing-book#31]
Currently, ymin/max are hard-coded for each plot, which makes the code poorly reusable. A suggestion would be to find the min/max from the data, and set the ymin/ymax to be: 
- `ymax = max + (max-min)/4`
- `ymin = min - (max-min)/4`

**Comment by zelenkastiot** [comment]
Hi @jcohenadad, yes I know. I plot them around the mean (plus/minus values). I will try to implement this

**Comment by zelenkastiot** [comment]
Hi @jcohenadad, I added two functions for calculating the limits on the y-label: 3641d9e6e283c466b401bc76bdb86a65d54e034d

If they look confusing, this is what they do: 
- Firstly the input is the matrix for that trace, which can be either: 
  - Single matrix for that trace (if there is only one metric displayed) 
  - Concatenated (double) matrix for that trace (if there is a comparison between two metrics) 
- Then, get rid of all the -100 ---> the -100 are added when I read the data. The reason for adding a random number (that is big) was to keep track of the real session number. Sessions were missing so when a session is missing I added this number to keep the number correct for the next one
- Then, calculate the `max` / `min` and pass them as arguments on the `y_axis_range` 

Sc of the functions: 
![image](https://user-images.githubusercontent.com/37374247/127850281-2097a256-c68d-4db7-8369-eaf48d5f0008.png)

**Comment by jcohenadad** [comment]
@zelenkastiot could you please comment directly in the code-- if someone reads the code they will not understand these tricks
things to comment:
- the 100 value trick
- the purpose of each function (a good habit in python is to use a triple quote to describe what the function does)

**Comment by jcohenadad** [comment]
Also: I realized the `get_limit_min/max()` code is duplicated across all plots, which is usually something to avoid in computer programming. Could you please remove this duplication (eg: create a series of functions used by all the plots in the book)

**Comment by zelenkastiot** [comment]
Sure thing! I will add the comments and remove the duplicate code fragments

**Comment by zelenkastiot** [comment]
Although, generally speaking, what if someone wants to run the second plot and all the functions are in the code cell for the first plot above. One has to go and fetch them one by one?

**Comment by zelenkastiot** [comment]
> Also: I realized the `get_limit_min/max()` code is duplicated across all plots, which is usually something to avoid in computer programming. Could you please remove this duplication (eg: create a series of functions used by all the plots in the book)

By this you mean to have a python script from which I import functions?

**Comment by jcohenadad** [comment]
> > Also: I realized the `get_limit_min/max()` code is duplicated across all plots, which is usually something to avoid in computer programming. Could you please remove this duplication (eg: create a series of functions used by all the plots in the book)
> 
> By this you mean to have a python script from which I import functions?

yes, for example. @mathieuboudreau mentioned it when he described what he did for his T1 mapping book.

**Comment by zelenkastiot** [comment]
Cool, I will try to implement this by tonight
